nav2_costmap_2d/plugins/costmap_filters/speed_filter.cpp

```diff
@@ -168,7 +168,7 @@ void SpeedFilter::maskCallback(
     filter_mask_.reset();
   }
 
-  filter_mask_ = msg;
+  std::atomic_store(&filter_mask_, std::make_shared<nav_msgs::msg::OccupancyGrid>(*msg));
```

Comment:
I could see this atomic store/load as sensible, but the extra copy over seems not. Keep that for the speed/keepout/binary